ext/src/main/java/io/burt/kafka/clients/KafkaClientsLibrary.java

```diff
@@ -58,6 +59,8 @@ static RubyClass mapErrorClass(Ruby runtime, Throwable t) {
       String javaName = t.getClass().getSimpleName();
       String rubyName = javaName.substring(0, javaName.length() - 9) + "Error";
       return (RubyClass) runtime.getClassFromPath(String.format("Kafka::Clients::%s", rubyName));
+    } else if (t instanceof TimeoutException) {
+      return (RubyClass) runtime.getModule("Timeout").getConstant("Error");
```

Comment:
I noticed that the Travis tests failed on 1.7 and it looks like this isn't available in JRuby 1.7, I think you need to `require 'timeout'` from the standard library first.